sentence_transformers/cross_encoder/evaluation/CEF1Evaluator.py

```diff
@@ -0,0 +1,128 @@
+
+        n_unique = np.unique(labels)
+        if n_unique == 2:
+            problem_type = "binary"
+        elif n_unique > 2:
+            problem_type = "multiclass"
+        else:
+            raise ValueError(
+                "Got only one distinct label. Please make sure there are at least two labels in the `labels` array."
+            )
+
+        if problem_type == "binary":
+            self.f1_callables = [
+                ("Binary F1 score", lambda x, y: f1_score(x, y, average="binary")),
+            ]
+        elif problem_type == "multiclass":
+            self.f1_callables = [
+                ("Macro F1 score", lambda x, y: f1_score(x, y, average="macro")),
+                ("Micro F1 score", lambda x, y: f1_score(x, y, average="micro")),
+                ("Weighted F1 score", lambda x, y: f1_score(x, y, average="weighted")),
+            ]
+
+        self.csv_file = "CEF1Evaluator" + (f"_{name}" if name else "") + "_results.csv"
+        self.csv_headers = ["epoch", "steps"] + [metric_name for metric_name, _ in self.f1_callables]
+
+    @classmethod
+    def from_input_examples(cls, problem_type: str, examples: List[InputExample], **kwargs):
+        sentence_pairs = []
+        labels = []
+
+        for example in examples:
+            sentence_pairs.append(example.texts)
+            labels.append(example.label)
+
+        return cls(problem_type, sentence_pairs, labels, **kwargs)
+
+    def __call__(
+        self,
+        model,
+        output_path: str = None,
+        epoch: int = -1,
+        steps: int = -1,
+    ) -> float:
+        if epoch != -1:
+            if steps == -1:
+                out_txt = f"after epoch {epoch}:"
+            else:
+                out_txt = f"in epoch {epoch} after {steps} steps:"
+        else:
+            out_txt = ":"
+
+        logger.info(f"CEF1Evaluator: Evaluating the model on {self.name} dataset {out_txt}")
+        pred_scores = model.predict(
+            self.sentence_pairs,
+            batch_size=self.batch_size,
+            show_progress_bar=self.show_progress_bar,
+            convert_to_numpy=True,
+        )
+        pred_labels = np.argmax(pred_scores, axis=1)
+
+        assert len(pred_labels) == len(self.labels)
+
+        save_f1 = []
+        for f1_name, f1_fn in self.f1_callables:
+            f1_val = f1_fn(pred_labels, self.labels)
+            save_f1.append(f1_val)
+            logger.info(f"{f1_name:20s}: {f1_val:.4f} \n")
+
+        if output_path is not None and self.write_csv:
+            csv_path = os.path.join(output_path, self.csv_file)
+            output_file_exists = os.path.isfile(csv_path)
+            mode = "a" if output_file_exists else "w"
+            with open(csv_path, mode=mode, encoding="utf-8") as f:
+                writer = csv.writer(f)
+                if not output_file_exists:
+                    writer.writerow(self.csv_headers)
+
+                writer.writerow([epoch, steps, *save_f1])
+
+        return save_f1[0]
```

Comment:
So we return either the binary F1 or the Macro F1? Perhaps we can write that in the docstring at the top (below `"The task type (binary or multiclass) ..."`)?